src/handlers.cpp

```diff
@@ -385,6 +398,36 @@ void ImpiTask::on_mar_response(Diameter::Message& rsp)
       break;
   }
 
+  // If not waiting for the assoicated IMPU put to succeed, tidy up now
+  if (!updating_assoc_public_ids)
+  {
+    delete maa;
```

Comment:
If we write directly to _maa and destroy it in this task's destructor, we can remove the need for `delete maa` here and elsewhere.